core/tee/tee_svc_cryp.c

```diff
@@ -2756,6 +2780,111 @@ static TEE_Result get_pbkdf2_params(const TEE_Attribute *params,
 }
 #endif
 
+#if defined(CFG_CRYPTO_SM2_KEP)
+static TEE_Result get_sm2_kep_params(const TEE_Attribute *params,
+				     uint32_t param_count,
+				     bool *is_initiator,
+				     struct ecc_public_key *peer_key,
+				     struct ecc_public_key *peer_eph_key,
+				     uint8_t **initiator_id,
+				     size_t *initiator_id_len,
+				     uint8_t **responder_id,
+				     size_t *responder_id_len,
+				     uint8_t **conf_in, size_t *conf_in_len,
+				     uint8_t **conf_out, size_t *conf_out_len)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	size_t n;
+	enum {
```

Comment:
I'm not so happy about such creative use of enums. I'd prefer something like:
``` C
enum {
   IS_INITIATOR,
   PEER_KEY_X,
   ...
};
```

And then use it as `found |= BIT(IS_INITIATOR)`

However, this is mostly a matter of taste so I leave that to you.

Reply:
Right, your approach seems less error-prone. I'll change this.